gobblin-modules/gobblin-eventhub/src/main/java/gobblin/eventhub/writer/EventhubDataWriter.java

```diff
@@ -0,0 +1,235 @@
+  }
+
+  /**
+   * Write a single record to eventhub
+   */
+  public WriteResponse write (JsonObject record) throws IOException {
+    long before = System.nanoTime();
+    String encoded = encodeRecord(record);
+    bytesWritten.mark(encoded.getBytes(Charsets.UTF_8).length);
+    request (encoded);
+    timer.update(System.nanoTime() - before, TimeUnit.NANOSECONDS);
+    return WriteResponse.EMPTY;
+  }
+
+  /**
+   * Total bytes that have been written
+   */
+  public long bytesWritten() {
+    return bytesWritten.getCount();
+  }
+
+  /**
+   * A signature which contains the duration.
+   * After the duration is expired, the signature becomes invalid
+   */
+  private void refreshSig () {
+    if (postStartTimestamp == 0 || (System.nanoTime() - postStartTimestamp) > Duration.ofMinutes(sigExpireInMinute).toNanos()) {
+      // generate signature
+      try {
+        signature = SharedAccessSignatureTokenProvider
+            .generateSharedAccessSignature(sasKeyName, sasKey, namespaceName, Duration.ofMinutes(sigExpireInMinute));
+        postStartTimestamp = System.nanoTime();
+        LOGGER.info ("Signature is refreshing: " + signature);
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  /**
+   * Send an encoded string to the Eventhub using post method
+   */
+  private int request (String encoded) throws IOException {
+    refreshSig();
+    HttpPost httpPost = new HttpPost(targetURI);
+    httpPost.setHeader("Content-type", "application/vnd.microsoft.servicebus.json");
+    httpPost.setHeader("Authorization", signature);
+    httpPost.setHeader("Host", namespaceName + ".servicebus.windows.net ");
+
+    StringEntity entity = new StringEntity(encoded);
+    httpPost.setEntity(entity);
+
+    CloseableHttpResponse response = httpclient.execute(httpPost);
+    StatusLine status = response.getStatusLine();
+    try {
+      LOGGER.info ("Return code is " + status.getStatusCode());
+      HttpEntity entity2 = response.getEntity();
+      // do something useful with the response body
+      // and ensure it is fully consumed
+      EntityUtils.consume(entity2);
+    } finally {
+      response.close();
+    }
+
+    return status.getStatusCode();
+  }
+
+  /**
+   * Each record of batch is wrapped by a 'Body' json object
+   * put this new object into an array, encode the whole array
+   */
+  private String encodeBatch (Batch<JsonObject> batch) throws IOException{
+    // Convert original json object to a new json object with format {"Body": "originalJson"}
+    // Add new json object to an array and send the whole array to eventhub using REST api
+    // Refer to https://docs.microsoft.com/en-us/rest/api/eventhub/send-batch-events
+    List<JsonObject> records = batch.getRecords();
+    ArrayList<EventhubRequest> arrayList = new ArrayList<>();
+
+    for (JsonObject record: records) {
+      LOGGER.info ("encoding record " + record.toString() + " from batch " + batch.getId());
```

Comment:
nit: LOGGER.info ("encoding record {}  from batch {}", record, batch.getId());

Further:
- logging this at info level is too verbose. 
- You have to be careful logging full records as it may expose sensitive information.